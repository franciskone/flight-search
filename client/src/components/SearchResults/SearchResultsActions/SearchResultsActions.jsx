import React from 'react';
import PropTypes from 'prop-types';
import BpkText from 'bpk-component-text';
import { withAlignment } from 'bpk-component-icon';
import { lineHeightLg, iconSizeLg } from 'bpk-tokens/tokens/base.es6';
import BpkLargePriceAlertIcon from 'bpk-component-icon/lg/price-alerts';

import { fakeOnClick, styleGetter } from '../../../util';
import STYLE from './SearchResultsActions.scss';

const c = styleGetter(STYLE);

export const actionItemOnKeyPress = action => evt =>
  evt.charCode === 13 && action();

const ActionItem = ({
  action, children, isLast = false,
}) => ( // TODO Franciskone: add snapshot test
  <div
    className={c(
      'SearchResultsActions__action-item',
        { 'SearchResultsActions__action-item--is-last': isLast },
      )
    }
    role="button"
    onClick={action}
    onKeyPress={actionItemOnKeyPress(action)}
    tabIndex="0"
  >
    <BpkText textStyle="lg">{children}</BpkText>
  </div>
);
ActionItem.propTypes = {
  action: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
  isLast: PropTypes.bool,
};
ActionItem.defaultProps = {
  isLast: false,
};

const AlignedPriceAlert = withAlignment(
  BpkLargePriceAlertIcon, lineHeightLg, iconSizeLg,
);

const SearchResultsActions = () => ( // TODO Franciskone: add snapshot test
  <div className={c('SearchResultsActions')}>
    <ActionItem action={() => fakeOnClick('Filter')}>
      Filter
    </ActionItem>
    
    <ActionItem action={() => fakeOnClick('Sort')}>
      Sort
    </ActionItem>
    
    <ActionItem action={() => fakeOnClick('Price Alert')} isLast>
      <AlignedPriceAlert
        className={c('SearchResultsActions__alert-icon')}
      />
      Price alerts
    </ActionItem>
  </div>
);

export default SearchResultsActions;

