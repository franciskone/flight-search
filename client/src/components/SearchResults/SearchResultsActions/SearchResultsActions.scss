@import '~bpk-mixins';

.SearchResultsActions {
  display: flex;
  padding: $bpk-spacing-sm;
  background-color: $bpk-color-white;
  color: $bpk-color-blue-500;

  &__action-list {
    display: flex;
  }
  &__action-item {
    &--right-padding {
      margin-right: $bpk-spacing-xxl;
    }
  }
  &__price-alert {

  }
}
