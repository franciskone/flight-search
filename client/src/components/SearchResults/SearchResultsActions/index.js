import SearchResultsActions from './SearchResultsActions';

export default SearchResultsActions;
