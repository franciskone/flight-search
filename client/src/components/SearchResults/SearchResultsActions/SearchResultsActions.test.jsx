import React from 'react';
import TestRenderer from 'react-test-renderer';

import SearchResultsActions, { actionItemOnKeyPress } from './SearchResultsActions';

describe('SearchResultsActions', () => {
  it('should render correctly', () => {
    const tree = TestRenderer.create(<SearchResultsActions />).toJSON();

    expect(tree).toMatchSnapshot();
  });
  
  it('actionItemOnKeyPress', () => {
    const testAction = jest.fn(() => null);
    const action = actionItemOnKeyPress(testAction);
    
    action({ charCode: 13 });
    action({ charCode: 0 });
  
    expect(testAction.mock.calls.length).toBe(1);
  });
});

