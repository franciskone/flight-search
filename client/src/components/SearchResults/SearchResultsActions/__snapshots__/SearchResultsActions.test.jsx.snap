// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultsActions should render correctly 1`] = `
<div
  className="SearchResultsActions"
>
  <div
    className="SearchResultsActions__action-item"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex="0"
  >
    <span
      className="bpk-text bpk-text--lg"
    >
      Filter
    </span>
  </div>
  <div
    className="SearchResultsActions__action-item"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex="0"
  >
    <span
      className="bpk-text bpk-text--lg"
    >
      Sort
    </span>
  </div>
  <div
    className="SearchResultsActions__action-item SearchResultsActions__action-item--is-last"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex="0"
  >
    <span
      className="bpk-text bpk-text--lg"
    >
      <span
        style={
          Object {
            "display": "inline-block",
            "lineHeight": "1.5rem",
            "marginTop": "0.1875rem",
            "verticalAlign": "top",
          }
        }
      >
        <svg
          className="SearchResultsActions__alert-icon"
          height="24"
          style={
            Object {
              "height": "1.5rem",
              "width": "1.5rem",
            }
          }
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18 13.2c.2 2.3 2 1.7 2 4 0 1.7-3.5 3.8-8 3.8s-8-2.1-8-3.8c0-2.3 1.8-1.7 2-4 .5-5.5 1-7.7 4.4-8.9C11.6 3.8 11 3 12 3s.4.8 1.6 1.3c3.5 1.2 3.9 3.4 4.4 8.9zM12 20c3.9 0 6.5-1.7 6.5-2.1 0-.7-2-2-6.5-1.9-4.5 0-6.5 1.4-6.5 2 0 .3 2.6 2 6.5 2zm-2.5-2.8c.7-.1 1.6-.2 2.5-.2s1.8.1 2.5.2c-.6 1.2-1.4 1.8-2.5 1.8s-2-.6-2.5-1.8z"
          />
        </svg>
      </span>
      Price alerts
    </span>
  </div>
</div>
`;
